docs/design/storage-auto-scale.md

```diff
@@ -0,0 +1,95 @@
+# Storage Automatic Scaler
+
+## Summary
+
+This feature would help users to get rid of maintenance task of keeping the system with enough available free space.
+In virtualized environments, we can scale the storage devices OSDs so this feature helps in automatic scaling. 
+
+## Proposal
+
+Implement auto storage scale CR per device class of a specific Storagecluster.
+
+## High-Level Design
+
+- Create a StorageAutoScale CR per Storagecluster device class
+    ```
+    apiVersion: ocs.openshift.io/v1
+    Kind: StorageAutoScaling
+    Metadata:
+        name: <storageclusterName>-<deviceClassName>
+        namespace: openshift-storage
+    Spec:
+        storageClusterName: <storageclusterName>
```

Comment:
Please use a LocalObjectReference structure for relationships within a single namespace

Reply:
I would disagree that a LocalObjectReference is necessary. Consider that PersistentVolumeClaims reference their bound PersistentVolume simply by the name. This allows for a more portable architecture that isn't possible when objects are bound strictly by their UID. An object spec copied from one cluster to another is portable using `storageClusterName` but isn't portable using a full, weighty LocalObjectReference

Reply:
(note: deleted prior comment where I had confused LocalObjectReference with the heavier-weight ObjectReference)